converters/toZigbee.js

```diff
@@ -841,234 +842,98 @@ const converters = {
-                    value.hue = utils.mapNumberRange(hsv.h, 0, 360, 0, 254);
-                    command = 'moveToHueAndSaturationAndBrightness';
-                }
-            } else if (value.hasOwnProperty('hsb')) {
-                let hsv = value.hsb.split(',').map((i) => parseInt(i));
-                newState.color = {h: hsv[0], s: hsv[1], v: hsv[2]};
-                hsv = utils.gammaCorrectHSV(utils.correctHue(hsv[0], meta), hsv[1], hsv[2]);
-                value.saturation = utils.mapNumberRange(hsv.s, 0, 100, 0, 254);
-                value.brightness = utils.mapNumberRange(hsv.v, 0, 100, 0, 254);
-                newState.brightness = value.brightness;
-                if (utils.getMetaValue(entity, meta.mapped, 'enhancedHue', 'allEqual', true)) {
-                    value.hue = utils.mapNumberRange(hsv.h % 360, 0, 360, 0, 65535);
-                    command = 'enhancedMoveToHueAndSaturationAndBrightness';
-                } else {
-                    value.hue = utils.mapNumberRange(hsv.h, 0, 360, 0, 254);
-                    command = 'moveToHueAndSaturationAndBrightness';
-                }
-            } else if (value.hasOwnProperty('h') && value.hasOwnProperty('s') && value.hasOwnProperty('v')) {
-                newState.color = {h: value.h, s: value.s, v: value.v};
-                const hsv = utils.gammaCorrectHSV(utils.correctHue(value.h, meta), value.s, value.v);
-                value.saturation = utils.mapNumberRange(hsv.s, 0, 100, 0, 254);
-                value.brightness = utils.mapNumberRange(hsv.v, 0, 100, 0, 254);
-                newState.brightness = value.brightness;
-                if (utils.getMetaValue(entity, meta.mapped, 'enhancedHue', 'allEqual', true)) {
-                    value.hue = utils.mapNumberRange(hsv.h % 360, 0, 360, 0, 65535);
-                    command = 'enhancedMoveToHueAndSaturationAndBrightness';
-                } else {
-                    value.hue = utils.mapNumberRange(hsv.h, 0, 360, 0, 254);
-                    command = 'moveToHueAndSaturationAndBrightness';
+            if (newColor.rgb !== null || newColor.xy !== null) {
+                // Convert RGB to XY color mode because at the moment all known devices support this mode.
+                newState.color_mode = constants.colorMode[1];
+                let xy;
+
+                if (newColor.rgb !== null) {
+                    const rgb = newColor.rgb;
+                    // XXX: store raw RGB, raw XY or corrected XY?
+                    newState.color = {
+                        red: utils.mapNumberRange(rgb.red, 0, 1, 0, 255),
+                        green: utils.mapNumberRange(rgb.green, 0, 1, 0, 255),
+                        blue: utils.mapNumberRange(rgb.blue, 0, 1, 0, 255),
+                    };
+                    xy = rgb.gammaCorrected().toXY();
+                } else /* xy */ {
+                    newState.color = xy.toObject(4);
+                    // TODO: add gamma correction
```

Comment:
I the color is already x/y I don't think we should do any gamma correction. So this todo can be removed.

Reply:
done